### PR TITLE
CEDS-2036 Accept only newer shape of declaration

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: app/uk/gov/hmrc/exports/models/declaration/Transport.scala [hunk 1]
```diff
@@ -27,7 +27,21 @@ case class Transport(
   meansOfTransportCrossingTheBorderNationality: Option[String] = None,
   meansOfTransportCrossingTheBorderType: Option[String] = None,
   meansOfTransportCrossingTheBorderIDNumber: Option[String] = None
-)
+) {
+  def hasBorderTransportDetails: Boolean = {
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/exports/models/declaration/Transport.scala]: :+1:

File: test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/consignment/ConsignmentBuilderSpec.scala [hunk 2]
```diff
@@ -70,21 +73,14 @@ class ConsignmentBuilderSpec extends WordSpec with Matchers with ExportsDeclarat
...
-            .buildThenAdd(refEq(Seq(Container("container", Seq(Seal("seal1"), Seal("seal2"))))), any[GoodsShipment.Consignment])
+          val consignment = goodsShipment.getConsignment
 
-          verify(departureTransportMeansBuilder)
-            .buildThenAdd(
-              refEq(DepartureTransport(borderModeOfTransportCode, meansOfTransportOnDepartureType, meansOfTransportOnDepartureIDNumber)),
-              any[Option[InlandModeOfTransportCode]],
-              any[GoodsShipment.Consignment]
-            )
+          println(consignment)
```
Comment (PauloPortugal) [inline test/unit/uk/gov/hmrc/exports/services/mapping/goodsshipment/consignment/ConsignmentBuilderSpec.scala]: Please remove

File: app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala [hunk 3]
```diff
@@ -191,13 +80,8 @@ object ExportsDeclarationRequest {
...
-
-  val bothReads: Reads[ExportsDeclarationRequest] = (__ \ "transport").readNullable[Transport].flatMap[ExportsDeclarationRequest] {
-    case Some(_) => readsVersion2
-    case None    => readsVersion1
-  }
+  ).apply(ExportsDeclarationRequest.apply _)
 
   implicit val format
-    : OFormat[ExportsDeclarationRequest] = OFormat(bothReads, Json.writes[ExportsDeclarationRequest]) // writes are used only for logging
+    : OFormat[ExportsDeclarationRequest] = OFormat(readsVersion2, Json.writes[ExportsDeclarationRequest]) // writes are used only for logging
```
Comment (PauloPortugal) [inline app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala]: I thought we were removing the explicit reads/writes.
@PatrykRudnicki  please confirm.

Comment (PatrykRudnicki) [reply app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala]: Output of the meeting was to come back to implementation before, without manually written reads and writes

**Comment by PatrykRudnicki** [comment]
@bambuchaAdm Scala formatter failed

File: app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala [hunk 4]
```diff
@@ -66,138 +62,9 @@ case class ExportsDeclarationRequest(
...
-  ).apply(ExportsDeclarationRequest.version2 _)
-
-  val bothReads: Reads[ExportsDeclarationRequest] = (__ \ "transport").readNullable[Transport].flatMap[ExportsDeclarationRequest] {
-    case Some(_) => readsVersion2
-    case None    => readsVersion1
-  }
-
   implicit val format
-    : OFormat[ExportsDeclarationRequest] = OFormat(bothReads, Json.writes[ExportsDeclarationRequest]) // writes are used only for logging
+    : OFormat[ExportsDeclarationRequest] = Json.format[ExportsDeclarationRequest] // writes are used only for logging
```
Comment (PatrykRudnicki) [inline app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala]: You can remove comment. It's not valid right now and there is no value to keep it

File: test/unit/uk/gov/hmrc/exports/models/declaration/ExportsDeclarationSpec.scala [hunk 5]
```diff
@@ -39,205 +39,15 @@ class ExportsDeclarationSpec extends WordSpec with MustMatchers {
...
-      |        "documentReference": "101SHIP2"
-      |      }
-      |    ]
-      |  },
-      |  "natureOfTransaction": {
-      |    "natureType": "1"
-      |  }
-      |}""".stripMargin
-
   val declarationVersion2: String =
```
Comment (PatrykRudnicki) [inline test/unit/uk/gov/hmrc/exports/models/declaration/ExportsDeclarationSpec.scala]: You can remove `2` from the name. There is one version of the declaration right now

File: app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala (hunk 4 above)
Comment (bambuchaAdm) [reply app/uk/gov/hmrc/exports/controllers/request/ExportsDeclarationRequest.scala]: It is. Changes in this writes though structure does not have effect on returned value by service.

**Pull request merged**